lib/parse.js

```diff
@@ -56,12 +61,151 @@ module.exports = {
         reason: 'Specification must contain a semantic version number of the API in the Info Object'
       };
     }
+    // Valid specification
+    return {
+      result: true,
+      openapi: spec
+    };
+  },
+
+  validateRoot: function (spec) {
+
+    // Checking for the all the required properties needed in a root file
+    if (!spec.hasOwnProperty('openapi')) {
+      return {
+        result: false,
+        reason: 'Specification must contain a semantic version number of the OAS specification'
+      };
+    }
+
+    if (!spec.hasOwnProperty('paths')) {
+      return {
+        result: false,
+        reason: 'Specification must contain Paths Object for the available operational paths'
+      };
+    }
+
+    if (!spec.hasOwnProperty('info')) {
+      return {
+        result: false,
+        reason: 'Specification must contain an Info Object for the meta-data of the API'
+      };
+    }
 
 
     // Valid specification
     return {
       result: true,
       openapi: spec
     };
+  },
+
+  getOasObject: function (file) {
+    let oasObj = file;
+
+    if (typeof oasObj === 'string') {
```

Comment:
else? All cases where getOasObject is being used, I see a string being passed to it. Why is this needed?